keanu-project/src/main/java/io/improbable/keanu/vertices/Vertex.java

```diff
@@ -155,40 +146,40 @@ public void setAndCascade(T value) {
      *
      * @param value the value to be observed
      */
+    @Override
     public void observe(T value) {
         this.value = value;
-        this.observed = true;
-    }
-
-    /**
-     * Cause this vertex to observe its own value, for example when generating test data
-     */
-    public void observeOwnValue() {
-        this.observed = true;
+        observation.observe(value);
     }
 
+    @Override
     public void unobserve() {
-        observed = false;
+        observation.unobserve();
     }
 
+    @Override
     public boolean isObserved() {
-        return observed;
+        return observation.isObserved();
+    }
+
+    public boolean matchesObservation() {
+        throw new UnsupportedOperationException();
     }
 
     public long getId() {
         return uuid;
     }
 
-    public Set<Vertex> getChildren() {
-        return children;
+    public List<Vertex<?>> getChildren() {
+        return ImmutableList.copyOf(children);
     }
 
     public void addChild(Vertex<?> v) {
         children.add(v);
     }
 
     public void setParents(Collection<? extends Vertex> parents) {
-        this.parents = new HashSet<>();
+        this.parents = new ArrayList<>();
```

Comment:
`parents` is now a list, so that the distribution backed vertices can assume that the order of their parameters is fixed